uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request-ctrl.js

```diff
@@ -151,15 +166,58 @@ define([
       })();
 
       /**
-       * Initializes leave request toil amounts
+       * Determines if the expiry date can be calculated based on the
+       * Number Of Days selected and the corresponding date field has value.
+       *
+       * @return {Boolean}
+       */
+      function canCalculateExpiryDate () {
+        return (vm.uiOptions.multipleDays && vm.request.to_date) ||
+          (!vm.uiOptions.multipleDays && vm.request.from_date);
+      }
+
+      /**
+       * Returns a promise with a date that can be used to calculate the expiry
+       * date. This date depends on the Multiple Days or Single Day options.
        *
        * @return {Promise}
        */
-      function loadToilAmounts () {
-        return OptionGroup.valuesOf('hrleaveandabsences_toil_amounts')
-          .then(function (amounts) {
-            vm.toilAmounts = _.indexBy(amounts, 'value');
+      function getReferenceDate () {
+        if (vm.uiOptions.multipleDays) {
+          return getReferenceDateForField({
+            hasErrors: !vm.request.to_date && !vm.request.from_date,
+            label: 'To Date',
+            value: vm.request.to_date
+          });
+        } else {
+          return getReferenceDateForField({
+            hasErrors: !vm.request.from_date,
+            label: 'From Date',
+            value: vm.request.from_date
           });
+        }
+      }
+
+      /**
+       * Returns a reference date using the field object as source.
+       * If the field has errors, it returns an error message.
+       * If the field has no value, it returns an empty message since it still
+       * is in the process of inserting values.
+       * And if everything is ok it returns the field's date value.
+       *
+       * @return {Promise}
+       */
+      function getReferenceDateForField (field) {
+        if (field.hasErrors) {
+          var message = 'Please select ' + field.label + ' to find expiry date';
```

Comment:
This message was present on the file before changes on line 51 as:

```
        if (!vm.request.from_date) {
          vm.errors = ['Please select from date to find expiry date'];

          return $q.reject(vm.errors);
        }
```

I updated it so if it throws an error message it would refer to the proper filed label.